src/ducc0/fft/fft.h

```diff
@@ -918,80 +986,84 @@ DUCC0_NOINLINE void general_nd(const cfmav<T> &in, vfmav<T> &out,
     if ((!plan) || (len!=plan->length()))
       plan = get_plan<Tplan>(len, in.ndim()==1);
 
-    execParallel(
-      util::thread_count(nthreads, in, axes[iax], fft_simdlen<T0>),
+    threading(
+      nthreads, in, axes[iax], fft_simdlen<T0>,
       [&](Scheduler &sched) {
         constexpr auto vlen = fft_simdlen<T0>;
         constexpr size_t nmax = 16;
         const auto &tin(iax==0? in : out);
-        multi_iter<nmax> it(tin, out, axes[iax], sched.num_threads(), sched.thread_num());
-        size_t nvec = 1;
-        if (it.critical_stride_trans(sizeof(T)))  // do bunches of transforms
-          nvec = nmax/vlen;
-        TmpStorage<T,T0> storage(in.size()/len, len, plan->bufsize(), nvec, inplace);
-
-        if (nvec>1)
+        ducc0::Range range;
+        while ((range = sched.getNext()))
           {
+          multi_iter<nmax> it(tin, out, axes[iax], range.lo, range.hi);
```

Comment:
looks like the diff is broken here and below - most of the change is just indentation since it's now wrapped in a while loop over `range = shed.getNext()`.